inf-terraform-azure/Jenkinsfile.template

```diff
@@ -0,0 +1,243 @@
+/* generated jenkins file used for building and deploying AWS-infrastructure in projects */
+
+def stackConfig
+def stackdeploy  = false
+
+def azureSubscriptionId
+def azureTenantId
+def azureClientId
+def azureClientSecret
+
+def azureSubscriptionIdTesting
+def azureTenantIdTesting
+def azureClientIdTesting
+def azureClientSecretTesting
+
+def workspaceKeyPrefix
+
+def tfvarsjsonfile
+def tfvarsjson
+
+
+@Library('ods-jenkins-shared-library@3.x') _
+
+import org.ods.services.ServiceRegistry
+import org.ods.services.BitbucketService
+import org.ods.util.GitCredentialStore
+
+
+odsComponentPipeline(
+  imageStreamTag: '@ods_namespace@/jenkins-agent-terraform:3.x',
+  branchToEnvironmentMapping: [
+    'master': 'dev',
+  ]
+) { context ->
+    stageGetConfig(context)
+    stageInit(context)
+    stageTest(context)
+    stagePlan(context)
+    stageDeploy(context)
+    stageSmoke(context)
+    stageReport(context)
+}
+
+def getEnvSettings(def tfBackend, def workspace ) {
+  def config = [
+    "TF_BACKEND=${tfBackend}"
```

Comment:
@matthiasbremser 
actually tfBackend is the environment in this case.
this is confusing

Reply:
I suppose there are multiple environments: runtime environment, workspace (environment), backend (of a particular environment). Does this make sense?

Reply:
hm, the value of tfBackend, so I'm concerned about the variable name itself.
On the other hand this Jenkinsfile.template will be superseded.